cirq/ops/linear_combinations.py

```diff
@@ -315,6 +318,63 @@ def copy(self) -> 'PauliSum':
         factory = type(self)
         return factory(self._linear_dict.copy())
 
+    def expectation(self,
+                    state: np.ndarray,
+                    qubit_map: Optional[Dict[raw_types.Qid, int]]=None
+                    ) -> float:
+        """Evaluate the expectation value of this PauliSum.
+
+        Compute the expectation value of this Hermitian PauliSum with respect to
+        an array representing either a wavefunction or density matrix.
+
+        See `PauliString.expectation` for input requirements.
+
+        Args:
+            state: An array representing a wavefunction or density matrix.
+            qubit_map: A map from all qubits used in this PauliSum to the
+            indices of the qubits that `state` is defined over.
+
+        Returns:
+            The expectation value of the input state.
+
+        Raises:
+            NotImplementedError if any term in this PauliSum is non-Hermitian.
+            ValueError if the input is a state with a size that is not a power
+            of 2, or a shape that is neither `(2 ** n,)` or `(2 ** n, 2 ** n)`.
+        """
+        if any([abs(p.coefficient.imag) > 0.0001 for p in self]):
+            raise NotImplementedError(
+                "Cannot compute expectation value of a non-Hermitian "
+                "PauliString <{}>. Coefficient must be real.".format(self))
+
+        # TODO: add support for mixtures.
```

Comment:
do, or open an issue and link here

Reply:
From discussion at the 8/21 sync and #1921 mixtures of states won't exist in cirq for the time being so I've removed this TODO